.circleci/config.yml

```diff
@@ -1,37 +1,319 @@
-# Javascript Node CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-javascript/ for more details
-#
+# Configuration file for https://circleci.com/gh/angular/flex-layout
+
+# Note: YAML anchors allow an object to be re-used, reducing duplication.
+# The ampersand declares an alias for an object, then later the `<<: *name`
+# syntax dereferences it.
+# See http://blog.daemonl.com/2016/02/yaml.html
+# To validate changes, use an online parser, eg.
+# http://yaml-online-parser.appspot.com/
+
+var_1: &docker_image angular/ngcontainer:0.7.0
+var_2: &cache_key v2-ng-layout-{{ .Branch }}-{{ checksum "yarn.lock" }}-0.7.0
+
+# Settings common to each job
+var_3: &job_defaults
+  working_directory: ~/ng
+  docker:
+    - image: *docker_image
+
+# Job step for checking out the source code from GitHub. This also ensures that the source code
+# is rebased on top of master.
+var_4: &checkout_code
+  checkout:
+    # After checkout, rebase on top of master. By default, PRs are not rebased on top of master,
+    # which we want. See https://discuss.circleci.com/t/1662
+    post: git pull --ff-only origin "refs/pull/${CI_PULL_REQUEST//*pull\//}/merge"
+
+# Restores the cache that could be available for the current Yarn lock file. The cache usually
+# includes the node modules and the build cache.
+var_5: &restore_cache
+  restore_cache:
+    key: *cache_key
+
+# Saves the cache for the current Yarn lock file. We store the node modules and the build
+# cache in order to make subsequent builds faster.
+var_6: &save_cache
+  save_cache:
+    key: *cache_key
+    paths:
+      - "node_modules"
+      - "~/dist/releases"
```

Comment:
Why would you want to cache the `release` output? The release output should be transferred through Circle's workspace storage.

Reply:
Will follow-up on this one offline